mega-zep-frontend/src/main/angular/frontend/src/app/modules/shared/services/error/error-handler.service.ts

```diff
@@ -1,37 +1,50 @@
-import {Injectable, Injector, ErrorHandler} from '@angular/core';
+import {Injectable, Injector, ErrorHandler, NgZone} from '@angular/core';
 import {LoggingService} from '../logging/logging.service';
-import {NotificationService} from '../notification/notification.service';
 import {ErrorService} from './error.service';
-import {HttpErrorResponse} from '@angular/common/http';
 import {configuration} from '../../constants/configuration';
+import {Router} from "@angular/router";
+import {UserService} from "../user/user.service";
 
 @Injectable({
   providedIn: 'root'
 })
 export class ErrorHandlerService implements ErrorHandler {
 
-  constructor(private injector: Injector) {
+  private readonly HTTP_STATUS_UNAUTHORIZED: number = 401;
+  private readonly HTTP_STATUS_FORBIDDEN: number = 403;
+
+  errorService: ErrorService;
 
+  constructor(private injector: Injector) {
```

Comment:
use of zone is dangerous and should be avoided, as mentioned above we should move the router to error-service to solve cyclic dependency